pyro/infer/traceenum_elbo.py

```diff
@@ -19,22 +19,23 @@
 from pyro.ops import packed
 from pyro.ops.contract import contract_tensor_tree, contract_to_tensor
 from pyro.poutine.enumerate_messenger import EnumerateMessenger
-from pyro.util import check_traceenum_requirements, warn_if_nan
+from pyro.util import check_traceenum_requirements, ignore_jit_warnings, warn_if_nan
 
 
 def _get_common_scale(scales):
     # Check that all enumerated sites share a common subsampling scale.
     # Note that we use a cheap weak comparison by id rather than tensor value, because
     # (1) it is expensive to compare tensors by value, and (2) tensors must agree not
     # only in value but at all derivatives.
-    scales_set = set()
-    for scale in scales:
-        if isinstance(scale, torch.Tensor) and scale.dim():
-            raise ValueError('enumeration only supports scalar poutine.scale')
-        scales_set.add(float(scale))
-    if len(scales_set) != 1:
-        raise ValueError("Expected all enumerated sample sites to share a common poutine.scale, "
-                         "but found {} different scales.".format(len(scales_set)))
+    with ignore_jit_warnings():
```

Comment:
nit: you could use the less invasive decorator form `@ignore_jit_warnings()` here.

Reply:
:+1: oh I didn't realize `contextlib2.contextmanager` back-ported Python 3's decorator syntax. I'll start using that more often now!